### DISCUSSION TITLE
SolaX "today" entities reset at 00:00 UTC not 00:00 local

### DISCUSSION THREAD

**Discussion by M0LTE** [wills106/homeassistant-solax-modbus#866] [Q&A]
I am in UK, which has daylight saving. Winter is UTC, summer is UTC+1. It is currently UTC+1.

Home Assistant shows graphs in local time, i.e. the rightmost point on the graph matches wallclock time, UTC+1.

Entities such as the following reset to zero at 01:00 UTC+1, i.e. 00:00 UTC:

sensor.solax_today_s_solar_energy
sensor.solax_today_s_export_energy
sensor.solax_battery_input_energy_today

Ideally they would reset at midnight local time.

What's the fix here? Is this an inverter setting, and if so, can this be compensated for in the integration?

**Comment by wills106** [comment]
Check the RTC sensor to see if it's correct in the Inverter.

Did you press the RTC sync button when the clocks changed the other week? On my Gen 3 you need to manually correct the time every month or so as it drifts out over time.

**Reply by M0LTE** [reply]
RTC sensor, just enabled, is returning UTC.

Just pressed Sync RTC, RTC sensor is still returning UTC.

So the inverter clock should be local?

**Reply by wills106** [reply]
It should match your HA time.
Is the timezone correct on HA? Europe/London?

**Reply by M0LTE** [reply]
It is correct- Time Zone is set to London in Settings->system->general, and the rightmost data point in graphs is the correct local time.

**Reply by M0LTE** [reply]
I have just unplugged and replugged the WiFi dongle from the inverter since it dropped off a few days ago, as it sometimes does, and the SolaX RTC sensor suddenly has started returning local time. With no other changes. Go figure.

(I am integrated via the Modbus connector, fwiw, nothing to do with the wifi.)

**Reply by M0LTE** [reply]
Hmm, something is up here.
I just pressed Sync RTC and it's changed the inverter back to UTC...
Set it manually in the SolaX app and the SolaX RTC sensor is displaying local time again.
